asv_bench/benchmarks/omnisci/benchmarks.py

```diff
@@ -81,6 +79,192 @@ def time_merge(self, shapes, how):
         )
 
 
+class TimeAppend:
+    param_names = ["shapes"]
+    params = [
+        BINARY_OP_DATA_SIZE[ASV_DATASET_SIZE],
+    ]
+
+    def setup(self, shapes):
+        self.df1 = generate_dataframe(
+            ASV_USE_IMPL, "int", *shapes[0], RAND_LOW, RAND_HIGH
+        )
+        self.df2 = generate_dataframe(
+            ASV_USE_IMPL, "int", *shapes[1], RAND_LOW, RAND_HIGH
+        )
+        trigger_import(self.df1, self.df2)
+
+    def time_append(self, shapes):
+        execute(self.df1.append(self.df2))
+
+
+class TimeBinaryOp:
+    param_names = ["shape", "binary_op"]
+    params = [
+        UNARY_OP_DATA_SIZE[ASV_DATASET_SIZE],
+        ["mul"],
+    ]
+
+    def setup(self, shape, binary_op):
+        self.df1 = generate_dataframe(ASV_USE_IMPL, "int", *shape, RAND_LOW, RAND_HIGH)
+        trigger_import(self.df1)
+
+    def time_mul_scalar(self, shape, binary_op):
+        execute(self.df1 * 2)
+
+    def time_mul_series(self, shape, binary_op):
+        execute(self.df1["col1"] * self.df1["col2"])
+
+    def time_mul_dataframes(self, shape, binary_op):
+        execute(self.df1 * self.df1)
+
+
+class TimeArithmetic:
+    param_names = ["shape"]
+    params = [
+        UNARY_OP_DATA_SIZE[ASV_DATASET_SIZE],
+    ]
+
+    def setup(self, shape):
+        self.df = generate_dataframe(ASV_USE_IMPL, "int", *shape, RAND_LOW, RAND_HIGH)
+        trigger_import(self.df)
+
+    def time_sum(self, shape):
+        execute(self.df.sum())
+
+    def time_median(self, shape):
+        execute(self.df.median())
+
+    def time_nunique(self, shape):
+        execute(self.df.nunique())
+
+    def time_apply(self, shape):
+        execute(self.df.apply(lambda df: df.sum()))
+
+    def time_mean(self, shape):
+        execute(self.df.mean())
+
+
+class TimeSortValues:
+    param_names = ["shape", "columns_number", "ascending_list"]
+    params = [
+        UNARY_OP_DATA_SIZE[ASV_DATASET_SIZE],
+        [1, 5],
+        [False, True],
+    ]
+
+    def setup(self, shape, columns_number, ascending_list):
+        self.df = generate_dataframe(ASV_USE_IMPL, "int", *shape, RAND_LOW, RAND_HIGH)
+        trigger_import(self.df)
+        self.columns = random_columns(self.df.columns, columns_number)
+        self.ascending = (
+            random_booleans(columns_number)
+            if ascending_list
+            else bool(random_booleans(1)[0])
```

Comment:
could you please explain the difference between `then` and `else` clauses of this `if`?

Reply:
It is possible to define the sort order for each column, if multiple columns are used for sorting.
In `then` clause returns list of bool values, in `else` clause only one bool value defining the order for all columns at once